JSR_Shopping_List_Functions-master/js/app.js

```diff
@@ -0,0 +1,947 @@
+
+// 	// TEST
+// 	describe('1. newShoppingListItem', () => {
+// 		it('should return an object with item and price attributes', () => {
+// 			const shoppingListItem = newShoppingListItem('test', 1)
+// 			chai.assert.equal(shoppingListItem.item, 'test');
+// 			chai.assert.equal(shoppingListItem.price, 1);
+// 		});
+// 	});
+
+
+// 	/* 2
+// 		@function addToShoppingList
+// 		@param item {object}
+// 		@param list {array, []}
+// 		@returns list
+// 		@description
+// 			add shoppinglist item object (ie:
+// 				{
+// 					'item': 'eggs',
+// 					'price': 1.59
+// 				}
+// 			) to a list
+// 			list is to DEFAULT to []
+// 			OPTIONAL:
+// 				- validate that the item is indeed a shoppingList item
+// 				- if shoppingList item is not passed in, throw error 
+// 	*/
+
+// 	// implement function here
+// const addToShoppingList = (item, list = []) => {
+// 	list.push(item);
+// 	return list;
+// }
+
+// console.log(addToShoppingList('eggs',['toast', 'carrots']));
+// 	// TEST
+// 	describe('2. addToShoppingList', () => {
+
+// 		it('should return a list', () => {
+// 			const newList = addToShoppingList({
+// 				'item': 'test',
+// 				'price': 1
+// 			});
+// 			chai.assert.equal(newList.length, 1)
+// 		});
+
+// 		it('should have one item that is a shopping list object', () => {
+// 			const newList = addToShoppingList({
+// 				'item': 'test',
+// 				'price': 1
+// 			});
+// 			chai.assert.equal(newList[0].item, 'test');
+// 			chai.assert.equal(newList[0].price, 1);
+// 		})
+// 	});
+
+// 	/* 3
+// 		@function removeFromShoppingList
+// 		@param list {array, []}
+// 		@returns list
+// 		@description
+// 			if array is empty, return it immediately
+// 			if array has items, remove the LAST item
+// 				and then return the array
+// 	*/
+
+// 	// implement function here
+// const removeFromShoppingList = (list) => {
+// 	if (list.length === 0){
+// 	return list;
+
+// 	}
+// 	else{
+// 		list.pop();
+// 		return list;
+// 	}
+// }
+// console.log(removeFromShoppingList([1,2,3,4,5]));
+
+// 	// TEST
+// 	describe('3. removeFromShoppingList', () => {
+// 		it('should remove from the end of the list', () => {
+// 			let list = addToShoppingList({
+// 				'item': 'test',
+// 				'price': 1
+// 			});
+// 			list = addToShoppingList({
+// 				'item': 'test2',
+// 				'price': 2
+// 			}, list);
+
+// 			// actually test function now
+// 			list = removeFromShoppingList(list);
+
+// 			chai.assert.equal(list.length, 1)
+// 			// asert only item in list is 'test' with price 1
+// 			chai.assert.equal(list[0].item, 'test')
+// 			chai.assert.equal(list[0].price, 1)
+// 		})
+// 	});
+
+// 	/* 4
+// 		@function removeFirstItem
+// 		@param list {array, []}
+// 		@returns list
+// 		@description
+// 			if array is empty, return it immediately
+// 			if array has items, remove the FIRST item
+// 				and then return the array
+// 	*/
+
+// 	// implement function here
+// 	const removeFirstItem = (list = [1,2,3,4,5]) =>{
```

Comment:
In this case, it would be better to submit your `list` default params as an empty array. It's better to start with the simplest case.